### ISSUE TITLE
Crash on 32 bit systems

### ISSUE THREAD

**Issue by patriotyk** [GENIVI/rvi_lib#28]
When invoking service on 32 bit architecture, I got segfault inside the jansson library. Crash happens when passing long long(64 bit)  to json_pack with parameter type "i" that uses int type and is 32 bit on 32 bit systems.

**Comment by tjamison** [comment]
@patriotyk, are you able to do some testing to determine whether this happens when using the jansson library directly? (i.e., is it a bug in the jansson library or a bug in how the library is invoked by rvi?)

**Comment by patriotyk** [comment]
It is bug in how the jansson library is invoked by lib_rvi, because it is wrong to pass 64bit integer to jasson and say to represent it as 32bit integer. But jansson should not crash in this situation the worst case we should get, it is truncated integer in parsed json. Also I have found that crash depends json structure. This is sample program that crashes:
```
#include <jansson.h>

char parameters[] = "[{\"device\":\"3edbffd6-af1a-41f8-8617-0eaed7cbc9e9\",\"services\":{\"abort\":\"genivi.org/3edbffd6-af1a-41f8-8617-0eaed7cbc9e9/abort\",\"chunk\":\"genivi.org/3edbffd6-af1a-41f8-8617-0eaed7cbc9e9/chunk\",\"finish\":\"genivi.org/3edbffd6-af1a-41f8-8617-0eaed7cbc9e9/finish\",\"getpackages\":\"genivi.org/3edbffd6-af1a-41f8-8617-0eaed7cbc9e9/getpackages\",\"notify\":\"genivi.org/3edbffd6-af1a-41f8-8617-0eaed7cbc9e9/notify\",\"start\":\"genivi.org/3edbffd6-af1a-41f8-8617-0eaed7cbc9e9/start\"},\"update_id\":\"1e42e4fa-1e6f-4152-92a3-55b85627d2e3\"}]";

int main(){
    
    json_t *params = json_loads( parameters, 0, NULL );
    if ( !params ){
        printf("cannot parse\n");
    }
    long long ttt = 1234;
    json_pack( 
            "{s:s, s:i, s:s, s:{s:s, s:i, s:o}}",
            "cmd", "rcv",
            "tid", 1, /* TODO: talk to Ulf about tid */
            "mod", "proto_json_rpc",
            "data", "service", "test",
                    "timeout", ttt,
                    "parameters", params
            );
return 0;
}
```

Replacing  ’s:i’ with ’s:I’ fixes this problem.

**Comment by tjamison** [comment]
Merged #29, closing issue
